src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -372,29 +388,186 @@ private static bool IsAnyFileSelected(IReadOnlyList<PackageItem> children)
                     DestinationPath = InstallationFolder.DestinationFolder,
                     Files = SelectedFiles.ToList()
-                });
-
-                await manifest.SaveAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+                };
 
                 EnvDTE.Project project = VsHelpers.DTE.SelectedItems.Item(1)?.ProjectItem?.ContainingProject;
                 project?.AddFileToProjectAsync(_configFileName);
 
-                await _libraryCommandService.RestoreAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+                await Shell.ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                RunningDocumentTable rdt = new RunningDocumentTable(Shell.ServiceProvider.GlobalProvider);
+                IVsTextBuffer document = rdt.FindDocument(Path.GetFullPath(_configFileName)) as IVsTextBuffer;
+
+                // The file isn't open. So we'll write to disk directly
+                if (document == null)
+                {
+                    manifest.AddLibrary(libraryInstallationState);
+
+                    await manifest.SaveAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+
+                    await _libraryCommandService.RestoreAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
 
-                _dispatcher.Invoke(() =>
+                    _dispatcher.Invoke(() =>
+                    {
+                        _closeDialog(true);
+                    });
+                }
+                else
                 {
-                    _closeDialog(true);
-                });
+                    // libman.json file is open, so we will write to the textBuffer.
+                    _dispatcher.Invoke(() =>
+                    {
+                        _closeDialog(true);
+                    });
+
+                    InsertIntoTextBuffer(document, libraryInstallationState);
+
+                    // Save manifest file so we can restore library files.
+                    rdt.SaveFileIfDirty(Path.GetFullPath(_configFileName));
+                }
             }
             catch { }
         }
 
+        private void InsertIntoTextBuffer(IVsTextBuffer document, LibraryInstallationState libraryInstallationState)
+        {
+            ITextBuffer textBuffer = GetTextBuffer(document);
+
+            if (textBuffer != null)
+            {
+                string insertionText;
+
+                JSONMember libraries = GetLibraries(textBuffer);
+
+                if (libraries != null && libraries.Children != null)
+                {
+                    JSONArray jsonArray = libraries.Children.OfType<JSONArray>().First();
+
+                    if (jsonArray != null)
+                    {
+                        string newLibrary = GetLibraryToBeInserted(libraryInstallationState);
+                        bool containsLibrary = jsonArray.BlockItemChildren.Count() > 0;
```

Comment:
>Count() [](start = 75, length = 7)

This is Linq, right? I'd just use .Any instead of this as there's no need to walk the whole enumeration